### PR TITLE
Feature/query formatter

### PR DESCRIPTION
**Pull request by HazelYuAhiru**
## Description
This PR improved the query syntax comparison textbox feature by adding a 'beautify query' button and nicely formatting the user input queries. This feature enhances the user experience and allows for easier comparison between two queries.

<p align="center">
Screen Shot 2023-10-10 at 11 26 02 AM
Before formatting the query
</p>

<p align="center">
Screen Shot 2023-10-10 at 11 25 04 AM
After formatting the query
</p>

## Changes Made
- Added "Beautify both queries" button to frontend interface.
- Imported the 'sql-formatter' library to beautify the original user input queries.
- Modified the 'updateMarkers' and 'findDistanceToWords' functions inside 'RewritingRuleModal.js' to support multi-line comparison after formatting the query.

### CODE REVIEW

File: client/src/dashboard/RewritingRuleModal.js [hunk 1]
```diff
@@ -182,11 +183,37 @@ const RewritingRuleModal = NiceModal.create(({user_id, rule=null, query=null}) =
     });
   };
 
-  function findDistanceToSpaces(inputString, currentPosition) {
+  const onBeautifyQuery = () => {
+    const sqlLanguage = (queryLanguage === 'pgsql') ? "postgresql" : queryLanguage;
```
Comment (seongjinyoon) [inline client/src/dashboard/RewritingRuleModal.js:187]: Can you clarify what this does?

Comment (HazelYuAhiru) [reply client/src/dashboard/RewritingRuleModal.js:187]: Do you mean function onBeautifyQuery? It is the function which uses sql-formatter library to re-format q0 and q1(beautify q0 and q1).

File: client/src/dashboard/RewritingRuleModal.js [hunk 2]
```diff
@@ -182,11 +183,39 @@ const RewritingRuleModal = NiceModal.create(({user_id, rule=null, query=null}) =
     });
   };
 
-  function findDistanceToSpaces(inputString, currentPosition) {
+  const onBeautifyQuery = () => {
+    // Beautify original query based on selected sql languages(mysql or pgsql)
+    const sqlLanguage = (queryLanguage === 'pgsql') ? "postgresql" : queryLanguage;
```
Comment (baiqiushi) [inline client/src/dashboard/RewritingRuleModal.js:188]: Can we make the initial list of options compatible with the query languages defined in sql-formater? Right now it seems only the `pgsql` is inconsistent with `postgresql`.

File: client/src/dashboard/RewritingRuleModal.js [hunk 3]
```diff
@@ -223,6 +311,7 @@ const RewritingRuleModal = NiceModal.create(({user_id, rule=null, query=null}) =
     //        1: {count: 30, added: true, removed: undefined, value: "this is the string being added"},
     //        2: {count: 33, added: undefined, removed: undefined, value: "this is the string without change"}]
     const diffs = diffChars(q0Line, q1Line);
+    console.log(diffs);
```
Comment (baiqiushi) [inline client/src/dashboard/RewritingRuleModal.js:314]: If we want to keep this console output, let's make it more informative. For example, console.log('[DEBUG] getDiffByLine:');
console.log(diffs);

File: client/src/dashboard/RewritingRuleModal.js (hunk 2 above)
Comment (HazelYuAhiru) [reply client/src/dashboard/RewritingRuleModal.js:188]: Regarding PostgreSQL, the Ace editor library (which I use for syntax highlighting and auto-completion) only accepts "pgsql" as the keyword, while the SQL formatter library only accepts "postgresql" as the keyword. I attempted to merge them into one but it did not work out so far :(

File: client/src/dashboard/RewritingRuleModal.js (hunk 3 above)
Comment (HazelYuAhiru) [reply client/src/dashboard/RewritingRuleModal.js:314]: Thank you so much! Fixed.

**Review by baiqiushi — APPROVED** [review]
LGTM.

**Pull request merged**